commit 0c802cbaa6db4203fc559829548e19ae5ee8c9c2
Author: Eric Sandeen <sandeen@redhat.com>
Date:   Mon Feb 4 08:54:27 2019 -0800

    xfs: fix inverted return from xfs_btree_sblock_verify_crc
    
    commit 7d048df4e9b05ba89b74d062df59498aa81f3785 upstream.
    
    xfs_btree_sblock_verify_crc is a bool so should not be returning
    a failaddr_t; worse, if xfs_log_check_lsn fails it returns
    __this_address which looks like a boolean true (i.e. success)
    to the caller.
    
    (interestingly xfs_btree_lblock_verify_crc doesn't have the issue)
    
    Signed-off-by: Eric Sandeen <sandeen@redhat.com>
    Reviewed-by: Christoph Hellwig <hch@lst.de>
    Reviewed-by: Darrick J. Wong <darrick.wong@oracle.com>
    Signed-off-by: Darrick J. Wong <darrick.wong@oracle.com>
    Signed-off-by: Luis Chamberlain <mcgrof@kernel.org>
    Signed-off-by: Sasha Levin <sashal@kernel.org>

diff --git a/fs/xfs/libxfs/xfs_btree.c b/fs/xfs/libxfs/xfs_btree.c
index 34c6d7bd4d18..bbdae2b4559f 100644
--- a/fs/xfs/libxfs/xfs_btree.c
+++ b/fs/xfs/libxfs/xfs_btree.c
@@ -330,7 +330,7 @@ xfs_btree_sblock_verify_crc(
 
 	if (xfs_sb_version_hascrc(&mp->m_sb)) {
 		if (!xfs_log_check_lsn(mp, be64_to_cpu(block->bb_u.s.bb_lsn)))
-			return __this_address;
+			return false;
 		return xfs_buf_verify_cksum(bp, XFS_BTREE_SBLOCK_CRC_OFF);
 	}
 
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1248_0c802cbaa6db4203fc559829548e19ae5ee8c9c2.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1248_0c802cbaa6db4203fc559829548e19ae5ee8c9c2.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
